spec/jobs/submissions_job_spec.rb

```diff
@@ -185,6 +185,182 @@
+              expect(new_test_run.test_group_results.ids).not_to eq(old_test_run.test_group_results.ids)
+            end
+          end
+        end
+
+        context 'for each new test group results' do
+          it 'creates the correct number of new test results' do
+            @new_submissions.zip(@original_submissions).each do |new_submission, old_submission|
+              new_submission.test_runs.zip(old_submission.test_runs).each do |old_test_run, new_test_run|
+                new_test_run.test_group_results.zip(old_test_run.test_group_results).each do |old_tgr, new_tgr|
+                  expect(new_tgr.test_results.size).to eq(old_tgr.test_results.size)
+                  expect(new_tgr.test_results.ids).not_to eq(old_tgr.test_results.ids)
+                end
+              end
+            end
+          end
+
+          it 'creates the correct number of new feedback files' do
+            @new_submissions.zip(@original_submissions).each do |new_submission, old_submission|
+              new_submission.test_runs.zip(old_submission.test_runs).each do |old_test_run, new_test_run|
+                new_test_run.test_group_results.zip(old_test_run.test_group_results).each do |old_tgr, new_tgr|
+                  expect(new_tgr.feedback_files.size).to eq(old_tgr.feedback_files.size)
+                  expect(new_tgr.feedback_files.ids).not_to eq(old_tgr.feedback_files.ids)
+                end
+              end
+            end
+          end
+        end
+      end
+    end
+
+    context 'for each new result that is created' do
+      context 'for remark data' do
+        let(:assignment) { create(:assignment_with_criteria_and_results_with_remark) }
+
+        it 'does not copy over remark information' do
+          expect(@new_results.all? { |result| result.remark_request_submitted_at.nil? }).to be(true)
+        end
+
+        it 'retains the marks from each original submission\'s original result' do
+          @new_submissions.zip(@original_submissions).each do |new_submission, old_submission|
+            # the last result in each submission is the original one in the old submission
+            expect(new_submission.current_result.marks.map(&:mark)).to eq(old_submission
+              .get_original_result.marks.map(&:mark))
+          end
+        end
+      end
+
+      context 'for marks' do
+        it 'creates the correct number of new marks for each result' do
+          @new_results.zip(@original_results).each do |new_result, old_result|
+            expect(new_result.marks.size).to eq(old_result.marks.size)
+            expect(new_result.marks.ids).not_to eq(old_result.marks.ids)
+          end
+        end
+
+        it 'retains the correct mark values for each result' do
+          @new_results.zip(@original_results).each do |new_result, old_result|
+            expect(new_result.marks.map(&:mark)).to eq(old_result.marks.map(&:mark))
```

Comment:
This is a bit flaky - 1% of the time the marks come out in different orders and causes the test to fail. Is sorting the arrays and comparing solely the content still a good test?